File: 박인우/[프로그래머스] 소수 만들기.py
def solution(nums):
    from itertools import combinations as cb
    answer = 0
    for a in cb(nums, 3):
        cand = sum(a)
        for j in range(2, int(cand**0.5)+1):
            if cand%j==0:
                break
        else:
            answer += 1
    return answer

File: 박인우/0207-0213/[프로그래머스] 소수 만들기.swift
import Foundation

func solution(_ nums:[Int]) -> Int {
    var answer = 0
    var sums = Set<Int>()
    
    for a in 0...nums.count-1 {
        if a == nums.count-1 {
            continue
        }
        for b in a+1...nums.count-1 {
            if b == nums.count-1 {
                continue
            }
            for c in b+1...nums.count-1 {
                let sum = nums[a] + nums[b] + nums[c]
                if check(sum) {
                    answer += 1
                }
            }
        }
    }
    
    return answer
}

func check(_ num: Int) -> Bool {
    for i in 2..<num {
        if num % i == 0 {
            return false
        }
    }
    return true
}
